lib/oli/delivery/sections.ex

```diff
@@ -2586,6 +2592,62 @@ defmodule Oli.Delivery.Sections do
     end)
   end
 
```

Comment:
This is a really critical part of the system. I'm not exactly sure what the intention of these
changes are, at the very least we need to see some unit tests around this change that clearly
demonstrate what it is trying to accomplish. These changes are also not listed in the original
ticket, so I think we need a better explaination in the PR/ticket of why these are being made

Reply:
This code was developed to remove the section resources associated with a deleted page. I've also incorporated a test to validate this action.

The function "build_reference_activity_ids" generates a list of IDs based on the hierarchy. Subsequently, within the function "build_published_resources_by_resource_id" (which is nested inside "create_nonstructural_section_resources"), it retrieves the publications related to this list of IDs. The condition "if length(processed_resource_ids) != 1" checks if there is more than one processed resource ID (implying that it's not just the curriculum resource itself), and if so, it proceeds to create the section resources.